### ISSUE TITLE
Add SikuliX as bundle

### ISSUE THREAD

**Issue by sriiora** [clearlinux/distribution#925]
I had followed the instructions in https://sikulix-2014.readthedocs.io/en/latest/basicinfo.html#sikulix-on-linux to setup Sikulix but fail. This is needed for UI test development on Clear Linux OS.

**Comment by lebensterben** [comment]
@sriiora 
Have you followed the instructions [here](https://sikulix-2014.readthedocs.io/en/latest/newslinux.html#version-1-1-4-special-for-linux-people)?

**Comment by sriiora** [comment]
> @sriiora
> Have you followed the instructions [here](https://sikulix-2014.readthedocs.io/en/latest/newslinux.html#version-1-1-4-special-for-linux-people)?

Yes I had followed, but I could not successfully compiled the tesseract library.

**Comment by lebensterben** [comment]
@sriiora 
Do you have `libleptonica` library installed? As far as I know it's not available via `swupd`.

**Comment by sriiora** [comment]
> @sriiora
> Do you have `libleptonica` library installed? As far as I know it's not available via `swupd`.
> 
> For testing,
> you can download a deb archive from here:
> https://packages.debian.org/sid/amd64/libleptonica-dev/download
> 
> Then extract it as a archive, and you'll see a `data.tar.xz`.
> Move contents of `./include` to `/usr/local/include`
> Move contents of `./usr/lib/x86_64-linux-gnu` to `/usr/lib64`.
> 
> Modify `/usr/lib64/pkgconfig/lept.pc` as the following
> 
> ```
> prefix=/usr
> exec_prefix=${prefix}
> libdir=${prefix}/lib64
> includedir=${prefix}/include/leptonica
> 
> Name: leptonica
> Description: An open source C library for efficient image processing and image analysis operations
> Version: 1.78.0
> Libs: -L${libdir} -llept
> Libs.private: -lz -lpng16 -lz -ljpeg -lgif -ltiff -lwebp
> Cflags: -I${includedir}
> ```

Thanks for the hint, I helps. Btw, I build leptonica from source instead of using the debian package. After successfully 'make' and 'make install', I manually did the following to steps:-
1. sudo cp -rf /usr/local/lib/* /usr/lib64
2. Modify `/usr/lib64/pkgconfig/lept.pc` as the following (minor different on 'includedir'
 
```
prefix=/usr
exec_prefix=${prefix}
libdir=${exec_prefix}/lib64
includedir=${prefix}/local/include/leptonica

Name: leptonica
Description: An open source C library for efficient image processing and image analysis operations
Version: 1.74.4
Libs: -L${libdir} -llept
Libs.private: -lz -lpng16 -lz -ljpeg  -ltiff -lwebp
Cflags: -I${includedir}
```

**Comment by lebensterben** [comment]
@sriiora So is SikuliX up and running now?

**Comment by sriiora** [comment]
Not yet test, missing wmctrl and libopencv_java. Do you know Clear Linux has these 2 packages ready?

**Comment by lebensterben** [comment]
@sriiora I'm away from my computer so I don't know for sure.

Try this, `swupd search wmctrl` and `swupd search opencv` and if you don't see any result, then find one for other system like Debian or Linux. And look for what are the files included in those packages. Then use `swupd search-file FILENAME` to look for that. If there's no search hit, then you need to compile them yourself, or try to use the prebuilt package for other distro.

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by sriiora** [comment]
For wmctrl I had downloaded the source and compiled without problem. But fail to compile libopencv_javaXYZ from source fail. I can compiled and produced all libopencv from source except libopencv_javaXYZ. 

I'm debugging the build configuration issue (miss match) generated by 'cmake'
```
/bin/ant: 第 160 行：cd: /bin/../share/ant/bin/..: 没有那个文件或目录
Error: Could not find or load main class org.apache.tools.ant.launch.Launcher
-- Could NOT find JNI (missing: JAVA_AWT_LIBRARY JAVA_JVM_LIBRARY JAVA_INCLUDE_PATH JAVA_INCLUDE_PATH2 JAVA_AWT_INCLUDE_PATH) 
```

```
--   OpenCV modules:
--     To be built:                 calib3d core dnn features2d flann highgui imgcodecs imgproc ml objdetect photo python2 python3 shape stitching superres ts video videoio videostab
```

```
--   Java:                          
--     ant:                         NO
--     JNI:                         NO
--     Java wrappers:               NO
--     Java tests:                  NO
```

I wish Clear Linux bundle also build and provide libopencv_javaXYZ as well since they already provide other libopencv. These libopencv* can be build from same source.

**Comment by lebensterben** [comment]
@sriiora 
As for `Error: Could not find or load main class org.apache.tools.ant.launch.Launcher`

Set the environment variable `export ANT_HOME=/usr/share/ant`

**Comment by AthenasJimenez** [comment]
@sriiora what is the current status? Do you still need help with this?

**Comment by sriiora** [comment]
I can build the library but it would be great if Clear Linux make it as one of the distributed bundle.
